rust/src/treefile.rs

```diff
@@ -1157,6 +1191,72 @@ pub(crate) struct LegacyTreeComposeConfigFields {
     pub(crate) automatic_version_prefix: Option<String>,
 }
 
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct DeriveCustom {
+    pub(crate) url: String,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) description: Option<String>,
+}
+
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct DeriveInitramfs {
+    pub(crate) regenerate: bool,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) etc: Option<Vec<String>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) args: Option<Vec<String>>,
+}
+
+/// These fields are only useful when deriving from a prior ostree commit;
+/// at the moment we only use them when translating an origin file
+/// to a treefile for client side assembly.
+#[derive(Serialize, Deserialize, Debug, Default)]
+#[serde(rename_all = "kebab-case")]
+pub(crate) struct DeriveConfigFields {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) from: Option<String>,
+
+    // Packages
```

Comment:
There's some interesting things going on here that are really an artifact of up until now the "origin" keyfiles being all about derivation and the treefile always being about "FROM scratch" i.e. non-derivation.

For example in this model plain `packages` are stored in the treefile.  As is `cliwrap` because those both already existed in the base treefile.